includes/class-amp-story-post-type.php

```diff
@@ -200,11 +208,46 @@ public static function filter_kses_allowed_html( $allowed_tags ) {
 	}
 
 	/**
-	 * Filter which styles will be printed on an AMP Story.
-	 *
-	 * At the moment, this will only allow Google fonts to be printed.
+	 * Filter which styles will be used in the edit page of an AMP Story.
 	 *
-	 * @todo We will need to allow sites to enqueue story-specific styles beyond fonts.
+	 * @param array $handles Style handles.
+	 * @return array Styles to print.
+	 */
+	public static function filter_editor_print_styles_array( $handles ) {
+		if (
+			! function_exists( 'get_current_screen' ) ||
+			! get_current_screen() ||
+			self::POST_TYPE_SLUG !== get_current_screen()->post_type ||
+			! get_current_screen()->is_block_editor
+		) {
+			return $handles;
+		}
+
+		return array_filter(
+			$handles,
+			function( $handle ) {
+				if ( ! isset( wp_styles()->registered[ $handle ] ) ) {
+					return false;
+				}
+				$dep = wp_styles()->registered[ $handle ];
+
+				// If we have amp-stories as dependency, allow the style.
+				if ( is_array( $dep->deps ) && in_array( self::AMP_STORIES_STYLE_HANDLE, $dep->deps, true ) ) {
+					return true;
+				}
+
+				// Disable the active theme's style.
+				if ( false !== strpos( $dep->src, get_stylesheet_directory_uri() ) ) {
```

Comment:
We should probably include a check for `get_template_directory_uri()` as well, in the case of parent/child themes.

Reply:
Added in ac852a3.